syz-ci/syz-ci.go

```diff
@@ -105,6 +105,9 @@ type Config struct {
 	// Path to upload coverage reports from managers (optional).
 	// Supported protocols: GCS (gs://) and HTTP PUT (http:// or https://).
 	CoverUploadPath string `json:"cover_upload_path"`
+	// Path to upload json coverage reports from managers (optional).
+	// Supported protocol: GCS (gs://)
```

Comment:
It looks like to enforce this restriction you need to write mode code (changes around getUploadURL/uploadFile). Why don't we want the same set of supported protocols that cover_upload_path has? It will be simpler to do and more consistent.

Reply:
It happened primarily because I wanted to set "mgr.cfg.PublishGCS" == false.

Reply:
We should keep PublishGCS as it is set now, at least until we provide a better alternative way to see coverage reports (and when we have the better way, we should just disable uploading of old reports).
I would add publish argument to uploadFile, pass PublishGCS value when we upload old coverage reports and pass false when we upload the new data.

Reply:
Ack. Let's check how it looks.
wip

Reply:
PTAL. It looks better than I expected.

Reply:
"Supported protocol: GCS (gs://)" is now outdated, right?

Reply:
It is the only tested scenario. It was the only reason for me to keep commend.
If it looks safe, I'll remove it.

Reply:
I would remove it. We use the same code used for cover_upload_path which does say that it supports http/https.

Reply:
Removing...

Reply:
Done.